sahi/model.py

```diff
@@ -663,3 +664,153 @@ def _create_object_prediction_list_from_original_predictions(
+
+        # Confirm model is loaded
+        assert self.model is not None, "Model is not loaded, load it by calling .load_model()"
+
+        from sahi.utils.torchvision import resize_image, numpy_to_torch
+        if self.image_size is not None:
+            image = resize_image(image, self.image_size)
+            image = numpy_to_torch(image)
+            prediction_result = self.model([image])
+
+        self._original_predictions = prediction_result
+
+    @property
+    def num_categories(self):
+        """
+        Returns number of categories
+        """
+        return len(self.category_mapping)
+
+    @property
+    def has_mask(self):
+        """
+        Returns if model output contains segmentation mask
+        """
+        has_mask = self.model.with_mask
+        return has_mask
+
+    @property
+    def category_names(self):
+        return self.category_mapping
+
+    def _create_object_prediction_list_from_original_predictions(
+            self,
+            shift_amount_list: Optional[List[List[int]]] = [[0, 0]],
+            full_shape_list: Optional[List[List[int]]] = None,
+    ):
+        """
+        self._original_predictions is converted to a list of prediction.ObjectPrediction and set to
+        self._object_prediction_list_per_image.
+        Args:
+            shift_amount_list: list of list
+                To shift the box and mask predictions from sliced image to full sized image, should
+                be in the form of List[[shift_x, shift_y],[shift_x, shift_y],...]
+            full_shape_list: list of list
+                Size of the full image after shifting, should be in the form of
+                List[[height, width],[height, width],...]
+        """
+        original_predictions = self._original_predictions
+        category_mapping = self.category_mapping
+
+        # compatilibty for sahi v0.8.15
+        if isinstance(shift_amount_list[0], int):
+            shift_amount_list = [shift_amount_list]
+        if full_shape_list is not None and isinstance(full_shape_list[0], int):
+            full_shape_list = [full_shape_list]
+
+        # parse boxes, masks, scores, category_ids from predictions
+        from sahi.utils.torchvision import classes
+        prediction_class = [classes[i] for i in list(original_predictions[0]['labels'].numpy())]
+        prediction_boxes = [[(i[0], i[1]), (i[2], i[3])] for i in
+                            list(original_predictions[0]['boxes'].detach().numpy())]
+        prediction_score = list(original_predictions[0]['scores'].detach().numpy())
+        prediction_thresh = [prediction_score.index(x) for x in prediction_score if x > self.confidence_threshold][-1]
+
+        boxes = prediction_boxes[:prediction_thresh + 1]
+        score = list(original_predictions[0]['scores'].detach().numpy())
+        classes = prediction_class[:prediction_thresh + 1]
+        """
```

Comment:
burayi simdilik test amacli mi kullaniyorsun?

Reply:
https://github.com/kadirnar/sahi/blob/torchvision/sahi/model.py#L762-L773 burasını sorduysanız hayır. 

Boxes, score ve classes değerlerini çekmek için 764-768 kodlarını yazmamız gerekiyor. Torchvision kendi colab dokümantasyonun da buna benzer kod yapısı kullanılmış. Eğer prediction = model(image) yapısı kullanılırsa daha az kod satırı ile işlem yapılabilir fakat o yapı ile ilgili çalışma daha yapmadım.

Reply:
@kadirnar 773te commentli kod blogu kalmis, ondan bahsediyordum

Reply:
Şuan tüm kodlar object detection için yazıldı. Segmentasyonu daha sonra yapacağım. O yüzden onu yorum satırı yaptım.